code/modules/jobs/job_types/job.dm

```diff
@@ -81,10 +81,8 @@
 	if(!H)
 		return FALSE
 	if(!visualsOnly)
-		var/datum/bank_account/bank_account = new(H.real_name, src)
-		bank_account.account_holder = H.real_name
-		bank_account.account_job = src
-		bank_account.account_id = rand(111111,999999)
+		var/obj/item/card/id/I = H.wear_id ? H.wear_id.GetID() : null
+		var/datum/bank_account/bank_account = new(I.registered_name, src)

```

Comment:
don't think this has handling for null values

Reply:
https://github.com/tgstation/tgstation/blob/0ebfc529f50bd6dd7f90ac311b3e97e787e6c8ca/code/modules/economy/account.dm#L9-L12